brave/src/main/java/brave/propagation/B3SinglePropagation.java

```diff
@@ -16,8 +16,12 @@
 import java.util.Collections;
 import java.util.List;
 
-/** Implements the propagation format described in {@link B3SingleFormat}. */
-public final class B3SinglePropagation<K> implements Propagation<K> {
+/**
+ * Implements the propagation format described in {@link B3SingleFormat}.
+ *
+ * @deprecated Since 5.9, use {@link B3Propagation#newFactoryBuilder()} to control inject formats.
+ */
+@Deprecated public final class B3SinglePropagation<K> implements Propagation<K> {
```

Comment:
Could remove the code here and delegate to the new stuff. The unit tests for this class still passing would give good confidence :)

Reply:
good idea

Reply:
the impl was package private so I just ripped out everything except the public ctor

Reply:
PS worthwhile trade of code!